kolibri/core/assets/src/components.js

```diff
@@ -149,6 +73,132 @@ var CrudCollection = Mn.CollectionView.extend({
 });
 
 
+var CrudAddItem = Mn.ItemView.extend({
+    // Template should either be the html *or* (as it is here) be a *function* that returns the rendered html.
+    // If it's a function, it's passed an object which is the "serialized" model of the view instance.
+    // In other words, if the view instance is given a model attribute when instantiated, then the argument
+    // to this function will be a copy of that model's .attributes hash.
+    template: function() {
+        var html =
+            '{{#unless valid}}<span>Invalid Input!</span><br />{{/unless}}' +
+            '{{#each create}}' +
+                '<input data-attr="{{ name }}" type="text" placeholder="{{ name }}" value="{{ default }}" /><br />' +
+            '{{/each}}' +
+            '<button class="create flat-button">Create</button>';
+        return Handlebars.compile(html)({create: this.create, valid: this.valid});
+    },
+
+    triggers: {
+        // Re-triggers the specified DOM event (from its template) as a Backbone.Event on this view.
+        'click .create': 'create'
+    },
+
+    initialize: function(options) {
+        // Some options when passed to the constructor are automatically bound to the view instance.
+        // This is the case for the "model" attribute, so passing a model to a view constructor means that the
+        // instance has it available automatically as "this.model".
+        // Custom options (like "create" below) must be bound to the view instance manually, or otherwise handled
+        // by the initialize function.
+        this.create = _.map(options.create, function(attr) {
+            return {
+                name: attr,
+                default: ''
+            };
+        });
+        this.validators = options.validators;
+        this.valid = true;
+        // Since "template" is a callback, bindAll ensures "this" refers the view instance.
+        _.bindAll(this, 'template');
+    },
+
+    // For event "xyz", you can specify a handler without having to set up a listener using the camel-cased
+    // "onXyz" convention.
```

Comment:
it is nice that it cuts down on boilerplate that would be necessary with vanilla Backbone though